lib/Connection.cs

```diff
@@ -56,6 +132,17 @@ public class Connection {
             }
         });
     }
+    
+    public void GetFileDirectory(string path) {
+        if (Directory.Exists(_currentPath + path))
+        {
+            Util.TaskRunSafe(() => SendMessage(new DirectoryRequest(path)));
+        }
+        else
+        {
+            Console.WriteLine("Wrong path");
+        }
+    }
```

Comment:
This is called on the side requesting a directory (client).
Why would a check if directory exists be here? It's meant to request a directory **on the server**, so why would you check if the same directory exists **on the client**.